### PR TITLE
feat: add publish job in template & use semantic-release to publish	

### PR DESCRIPTION
**Pull request by minz1027**
Add commit message format to CONTRIBUTING.md. 
Relates to: https://github.com/screwdriver-cd/screwdriver/issues/373

### CODE REVIEW

**Comment by stjohnjohnson** [comment]
I don't think this is a doc update.

Also can you fix generator-screwdriver to use the new feature?

**Comment by minz1027** [comment]
@stjohnjohnson so there will be a publish job in screwdriver.yaml template by default?

**Comment by stjohnjohnson** [comment]
I thought there was, @minz1027 :(  Yes plz

**Comment by minz1027** [comment]
@stjohnjohnson fixed! can you re-review it?

File: app/templates/screwdriver.yaml [hunk 1]
```diff
@@ -1,6 +1,21 @@
...
+
+jobs:
+    main:
+        steps:
+            - install: npm install
+            - test: npm test
+
+    publish:
+        steps:
+            - install: npm install semantic-release
```
Comment (stjohnjohnson) [inline app/templates/screwdriver.yaml]: @petey do you think this is going to be a problem given no restriction?

**Pull request merged**